### PR TITLE
Minor ffi refresh

### PR DESCRIPTION
**Pull request by NiLuJe**
I wanted to double-check that stuff would be sane for AArch64, and that no broken type conversions had sneaked in somewhere...

(Related to https://github.com/koreader/ffi-cdecl/pull/8)

### CODE REVIEW

File: thirdparty/dropbear/CMakeLists.txt [hunk 1]
```diff
@@ -15,30 +15,30 @@ assert_var_defined(CHOST)
...
+    set(PATCH_CMD7 "${KO_PATCH} ${CMAKE_CURRENT_SOURCE_DIR}/dropbear-2018.76-scp-command-hack-kobo.patch")
 elseif($ENV{KINDLE})
-    set(PATCH_CMD7 sh -c "${KO_PATCH} ${CMAKE_CURRENT_SOURCE_DIR}/dropbear-2018.76-scp-command-hack-kindle.patch")
+    set(PATCH_CMD7 "${KO_PATCH} ${CMAKE_CURRENT_SOURCE_DIR}/dropbear-2018.76-scp-command-hack-kindle.patch")
 elseif($ENV{CERVANTES})
-    set(PATCH_CMD7 sh -c "${KO_PATCH} ${CMAKE_CURRENT_SOURCE_DIR}/dropbear-2018.76-scp-command-hack-cervantes.patch")
+    set(PATCH_CMD7 "${KO_PATCH} ${CMAKE_CURRENT_SOURCE_DIR}/dropbear-2018.76-scp-command-hack-cervantes.patch")
 else()
-    set(PATCH_CMD7 sh -c "true")
+    set(PATCH_CMD7 "true")
```
Comment (Frenzie) [inline thirdparty/dropbear/CMakeLists.txt]: Does CMake error without this else branch?

Comment (NiLuJe) [reply thirdparty/dropbear/CMakeLists.txt]: Probably not, that's probably a leftover or a c/p mistake from where those where handled in shell chain (i.e., PATCH1 && PATCH2).

File: thirdparty/glib/CMakeLists.txt [hunk 2]
```diff
@@ -24,18 +24,18 @@ ep_get_binary_dir(BINARY_DIR)
 # sometimes just hoping there's a python binary somewhere in $PATH (which, granted, there should be)...
 # Enforce that to python3 for containers where that's the only available Python version.
 if(($ENV{APPIMAGE}) OR ($ENV{CERVANTES}) OR ($ENV{KINDLE}) OR ($ENV{KOBO}))
-    set(PATCH_CMD1 sh -c "${KO_PATCH} ${CMAKE_CURRENT_SOURCE_DIR}/glib-2.56.2-python3.patch")
+    set(PATCH_CMD1 "${KO_PATCH} ${CMAKE_CURRENT_SOURCE_DIR}/glib-2.56.2-python3.patch")
 else()
-    set(PATCH_CMD1 sh -c "true")
+    set(PATCH_CMD1 "true")
```
Comment (Frenzie) [inline thirdparty/glib/CMakeLists.txt]: Another one.

File: thirdparty/glib/CMakeLists.txt [hunk 3]
```diff
@@ -24,18 +24,18 @@ ep_get_binary_dir(BINARY_DIR)
...
 endif()
 # Android makes Linux syscalls hard to get to... Fix pilfered from Python.
 if($ENV{ANDROID})
-    set(PATCH_CMD2 sh -c "${KO_PATCH} ${CMAKE_CURRENT_SOURCE_DIR}/glib-2.58.2-android-linux-syscalls.patch")
+    set(PATCH_CMD2 "${KO_PATCH} ${CMAKE_CURRENT_SOURCE_DIR}/glib-2.58.2-android-linux-syscalls.patch")
 else()
-    set(PATCH_CMD2 sh -c "true")
+    set(PATCH_CMD2 "true")
```
Comment (Frenzie) [inline thirdparty/glib/CMakeLists.txt]: ^

File: thirdparty/glib/CMakeLists.txt [hunk 4]
```diff
@@ -49,7 +49,7 @@ set(gen_arm_config "${gen_arm_config}ac_cv_func_posix_getgrgid_r=no\\n")
 set(gen_arm_config "${gen_arm_config}ac_cv_func_getpwuid_r=no\\n")
 set(gen_arm_config "${gen_arm_config}ac_cv_func_getgrgid_r=no\\n")
 set(gen_arm_config "${gen_arm_config}ac_cv_func_timegm=no\\n")
-set(gen_arm_config sh -c "printf \"${gen_arm_config}\" > arm_cache.conf")
+set(gen_arm_config "printf \"${gen_arm_config}\" > arm_cache.conf")
```
Comment (Frenzie) [inline thirdparty/glib/CMakeLists.txt]: Huh, for some reason I thought that was a shell builtin but apparently not.

```
$ which printf
/usr/bin/printf
```

Comment (NiLuJe) [reply thirdparty/glib/CMakeLists.txt]: It's both ;).

Comment (Frenzie) [reply thirdparty/glib/CMakeLists.txt]: I see… well, in any case I don't have to go waving my hands saying it won't work without sh -c. :-P

**Comment by NiLuJe** [comment]
> That script doesn't have the executable bit though (unless you changed that in one of the commits here)

I did (596edc8) ;).

File: thirdparty/turbo/CMakeLists.txt [hunk 5]
```diff
@@ -12,9 +12,9 @@ assert_var_defined(LDFLAGS)
 
 ep_get_source_dir(SOURCE_DIR)
 
-set(BUILD_CMD sh -c "${KO_MAKE_RECURSIVE} CC=\"${CC}\" CFLAGS=\"${CFLAGS}\" LDFLAGS=\"${LDFLAGS}\" all")
+set(BUILD_CMD "${KO_MAKE_RECURSIVE} CC=\"${CC}\" CFLAGS=\"${CFLAGS}\" LDFLAGS=\"${LDFLAGS}\" all")
 
-set(PATCH_CMD "${ISED} 's|CC=|CC:=|' ${SOURCE_DIR}/Makefile")
+set(PATCH_CMD sh -c "${ISED} 's|CC=|CC:=|' ${SOURCE_DIR}/Makefile")
```
Comment (Frenzie) [inline thirdparty/turbo/CMakeLists.txt]: ??

Comment (NiLuJe) [reply thirdparty/turbo/CMakeLists.txt]: It's used to build a shell one-liner below on Android ;).

**Pull request merged**